### PR TITLE
Use deployed play routes compiler cli

### LINKED ISSUE #5: rules_scala or rules_scala_annex?
The README says this project is for use with [rules_scala](https://github.com/bazelbuild/rules_scala) but the [play-routes-compiler/BUILD.bazel](https://github.com/lucidsoftware/rules_play_routes/blob/master/play-routes-compiler/BUILD.bazel#L1) file uses [rules_scala_annex](https://github.com/andyscott/rules_scala_annex). Which set of Scala build rules are expected to work?

### PR DESCRIPTION
**Pull request by SrodriguezO**
This PR doesn't make the compiler configurable yet--that'll come in a second PR. This PR makes it s.t. we no longer leak the dependency on higherkindness/rules_scala.

This closes https://github.com/lucidsoftware/rules_play_routes/issues/5

Next steps:
- Move the tests into their own workspace
- Make the compiler configurable

### CODE REVIEW

**Comment by SrodriguezO** [comment]
@jjudd @kojustin 
Lmk what you think. I moved the compiler cli to https://github.com/lucidsoftware/play_routes_compiler_cli and published the current version to Maven. 

This should be enough to remove the dependency on `higherkindness/rules_scala` for users of `lucidsoftware/rules_play_routes`, but I'll also move the tests into their own workspace as James suggested.

**Comment by SrodriguezO** [comment]
I thought about it a bit more over the weekend and I'm leaning towards moving the tests to their own workspace in a separate PR. It's not as trivial as I'd anticipated, and I don't want to block this PR on that. It's also not strictly tied to what this PR is doing. @jjudd wdyt?

File: WORKSPACE
Comment (jjudd) [inline WORKSPACE:129]: Can you add a TODO here to move this to a test workspace? Thanks

**Review by jjudd — APPROVED** [review]

**Comment by jjudd** [comment]
LGTM pending the TODO comment

**Review by borkaehw — APPROVED** [review]
LGTM

**Pull request merged**
